### PR TITLE
366 process team work

### LINKED ISSUE #366: Process - Team Work
_Show a screenshot of your project board right before hand-in. Briefly describe which tasks are still unresolved, i.e., which features are missing from your applications or which functionality is incomplete._

_Briefly describe and illustrate the flow of activities that happen from the new creation of an issue (task description), over development, etc. until a feature is finally merged into the main branch of your repository._

### PR DESCRIPTION
**Pull request by kartoffelmozart**
pls tell me what else is needed on this one

### CODE REVIEW

File: docs/report.md [hunk 1]
```diff
@@ -40,7 +31,30 @@ Our domain model consists of two data entities, which depict the attributes of a
...
+- Backlog: An issue in the code that has no immediate solution; the group must talk about it and figure out how to turn this into an issue ready for implementation OR an issue that has a concrete solution, but should not be focused on yet, as the code base is not ready or the issue is not relevant to work with at the moment
+- Ready: Issues ready for a team member to assign themselves and being implementation.
+
+A team member then creates a GitHub branch with the name of the issue and starts on implementing the change described.
+Often we have worked on larger/more fundemental changes in groups of 2-3 or as the whole team (mob programming).
+We have (with varying consistency) written the code for an issue in smaller commits of working code.
+When the code for an issue is done, the code is pushed to the online repository on its branch. A pull request is made, the code is reviewed by one or more team members, different from the author of the code. The reviewer may then decide if more work is needed from the issue asignee, or if the change is ready to merge with the main branch.
+Upon merge, the issue is deleted.
+Below is a user flow diagram showing the process of an issue:
```
Comment (palleBrandt) [inline docs/report.md:56]:
```suggestion
See figure \ref{The life of an issue}, a user flow diagram showing the process of an issue:
```

File: docs/report.md [hunk 2]
```diff
@@ -14,18 +14,9 @@ numbersections: true
...
-
-### Author
-An Author represents the user in our program. The user can write posts (Cheeps), which is why they are named Author. An Author has references to all their Cheeps, as well as their name and email. It is possible to follow an Author; therefore, the Author also has references to the Authors who follow it and the Authors it follows.
-
-Our domain model consists of two data entities, which depict the attributes of a 'Cheep' and an 'Author' in the context of our application. Figure \ref{domainModelImage} shows the two classes, 'Cheep' and 'Author,' with their fields and cardinality relationships between each other and within themselves. The 'Cheep' class has a one-to-one relationship with an 'Author' class, and an 'Author' class has a zero-to-many relationship with the 'Cheep' class. Furthermore the 'Author' class has a zero-to-many relationship with authors who follow it and a zero-to-many relationship with authors whom it follows.
-
-![Illustration of the _Chirp!_ data model as UML class diagram.\lable{domainModelImage}](https://raw.githubusercontent.com/ITU-BDSA23-GROUP12/Chirp/main/docs/images/DomainModel.drawio1.png)
+Here comes a description of our domain model.
```
Comment (palleBrandt) [inline docs/report.md:17]:
```suggestion
Our Application builds on two main consepts: Cheeps and Authors.

### Cheep
A Cheep is the understanding of a post, which holds a message from the author of the post. It has a refernce to its author and a knowlegde of when it was posted.

### Author
An Author represents the user in our program. The user can write posts (Cheeps), which is why they are named Author. An Author has references to all their Cheeps, as well as their name and email. It is possible to follow an Author; therefore, the Author also has references to the Authors who follow it and the Authors it follows.

Our domain model consists of two data entities, which depict the attributes of a 'Cheep' and an 'Author' in the context of our application. Figure \ref{domainModelImage} shows the two classes, 'Cheep' and 'Author,' with their fields and cardinality relationships between each other and within themselves. The 'Cheep' class has a one-to-one relationship with an 'Author' class, and an 'Author' class has a zero-to-many relationship with the 'Cheep' class. Furthermore the 'Author' class has a zero-to-many relationship with authors who follow it and a zero-to-many relationship with authors whom it follows.

![Illustration of the _Chirp!_ data model as UML class diagram.\lable{domainModelImage}](https://raw.githubusercontent.com/ITU-BDSA23-GROUP12/Chirp/main/docs/images/DomainModel.drawio1.png)
```

File: docs/report.md [hunk 3]
```diff
@@ -14,18 +14,9 @@ numbersections: true
...
 
+![Illustration of the _Chirp!_ data model as UML class diagram.](docs/images/domain_model.png)
```
Comment (palleBrandt) [inline docs/report.md:19]:
```suggestion
```

**Review by palleBrandt — CHANGES_REQUESTED** [review]
Very nice section.

I have added some minor suggestions for you to approve:
- [x] We have accidentally deleted Marcus part about domainmodel, therefore i have suggested to reinsert it.
- [x] A reference to the figure number.

**Review by RazzerTazzer — COMMENTED** [review]
I like the diagram alot. Im a little confused about backlog and if it fails the 'Ready for implementation'. If no, it seemss a new issue would be created. To me it makes more sense to have the arrow point to backlog again.

File: docs/report.md [hunk 4]
```diff
@@ -40,7 +31,30 @@ Our domain model consists of two data entities, which depict the attributes of a
...
+- A character count above the cheep box, denoting how many characters can yet be used, up to 160.
+- Profile pictures, fetched from GitHub.
+- Functionality to reference other users in a cheep, and a page with the newest cheeps in which the user is tagged.
+- Emoticon reactions to cheeps, such as happy face or tractor.
+- Comments under cheeps.
+- A link redirecting to a user's GitHub page.
+
+#### Work flow
+We have implemented the code in short incremental steps as, starting with a GitHub issue, describing a desired change in the code, who would benefit from this change, a more detailed description and a definition of done.
+The issues are created to match the project description provided by our professors Helge and Rasmus, or to match changes and details that we as a group or individually found relevant to implement.
```
Comment (allanzigge) [inline docs/report.md:45]: fluff
```suggestion
The issues are created to match the project description provided by our professors, or to match changes and details that we as a group or individually found relevant to implement.
```

File: docs/report.md [hunk 5]
```diff
@@ -40,7 +31,30 @@ Our domain model consists of two data entities, which depict the attributes of a
...
+On creation, an issue is placed in one of three categories:
+- To-do: a place for issues that contain an overall idea for the group to think on, not something that should be implemented as code. 
+- Backlog: An issue in the code that has no immediate solution; the group must talk about it and figure out how to turn this into an issue ready for implementation OR an issue that has a concrete solution, but should not be focused on yet, as the code base is not ready or the issue is not relevant to work with at the moment
+- Ready: Issues ready for a team member to assign themselves and being implementation.
+
+A team member then creates a GitHub branch with the name of the issue and starts on implementing the change described.
```
Comment (allanzigge) [inline docs/report.md:51]:
```suggestion
When a team member takes responsibility for an issue, he creates a branch linked to the issue with the name of the issue and starts on implementing the change described.
```

File: docs/report.md [hunk 6]
```diff
@@ -40,7 +31,30 @@ Our domain model consists of two data entities, which depict the attributes of a
...
+Often we have worked on larger/more fundemental changes in groups of 2-3 or as the whole team (mob programming).
```
Comment (allanzigge) [inline docs/report.md:52]:
```suggestion
Often we have worked on larger/more fundemental changes in groups of 2-3 or as the whole team (mob-/pair programming).
```

File: docs/report.md [hunk 7]
```diff
@@ -40,7 +31,30 @@ Our domain model consists of two data entities, which depict the attributes of a
...
+We have (with varying consistency) written the code for an issue in smaller commits of working code.
+When the code for an issue is done, the code is pushed to the online repository on its branch. A pull request is made, the code is reviewed by one or more team members, different from the author of the code. The reviewer may then decide if more work is needed from the issue asignee, or if the change is ready to merge with the main branch.
+Upon merge, the issue is deleted.
```
Comment (allanzigge) [inline docs/report.md:55]:
```suggestion
Upon merge, the issue is closed and moved to Done in our project board.
```

File: docs/report.md [hunk 8]
```diff
@@ -40,7 +31,30 @@ Our domain model consists of two data entities, which depict the attributes of a
...
+Below is a user flow diagram showing the process of an issue:
+![The life of an issue](https://github.com/ITU-BDSA23-GROUP12/Chirp/blob/366-process-team-work/docs/images/issue%20process.drawio.png)
```
Comment (allanzigge) [inline docs/report.md:57]:
```suggestion
![The life of an issue](https://github.com/ITU-BDSA23-GROUP12/Chirp/blob/366-process-team-work/docs/images/issue%20process.drawio.png?raw=true)
```

**Review by allanzigge — CHANGES_REQUESTED** [review]

File: docs/report.md (hunk 2 above)
Comment (kartoffelmozart) [reply docs/report.md:17]: Den her del hører vist ikke til team work sektionen, muligvis det er bedre at ændre de ting på dens egen branch

Comment (Maquidos) [reply docs/report.md:17]: tror det er fordi at der vil være merge konflikt hvis ikke det er med

**Review by Maquidos — COMMENTED** [review]
looks nice, with the things implemented as suggested from the others.

**Review by palleBrandt — APPROVED** [review]
Looks good with the new changes

**Review by allanzigge — APPROVED** [review]

File: docs/images/issue process.drawio.png
Comment (allanzigge) [inline docs/images/issue process.drawio.png:1]: Need a few changes

**Review by allanzigge — CHANGES_REQUESTED** [review]

**Review by allanzigge — APPROVED** [review]

**Pull request merged**